### DISCUSSION TITLE
Can these warnings be removed?

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD#3291] [General]
I've found myself ignoring warnings when I shouldn't...

In that regard, it would be helpful to fix some warnings that are always there, so I pay more attention...

Can these be fixed?

```
[WARNING STA-0164] /OpenROAD-flow-scripts/flow/platforms/asap7/lib/asap7sc7p5t_SIMPLE_RVT_TT_nldm_211120.lib.gz line 13178, timing group from output port.
[WARNING STA-0164] /OpenROAD-flow-scripts/flow/platforms/asap7/lib/asap7sc7p5t_SIMPLE_RVT_TT_nldm_211120.lib.gz line 13211, timing group from output port.
[WARNING STA-0164] /OpenROAD-flow-scripts/flow/platforms/asap7/lib/asap7sc7p5t_SIMPLE_RVT_TT_nldm_211120.lib.gz line 13244, timing group from output port.
[WARNING STA-0164] /OpenROAD-flow-scripts/flow/platforms/asap7/lib/asap7sc7p5t_SIMPLE_RVT_TT_nldm_211120.lib.gz line 13277, timing group from output port.
[WARNING STA-0164] /OpenROAD-flow-scripts/flow/platforms/asap7/lib/asap7sc7p5t_SIMPLE_RVT_TT_nldm_211120.lib.gz line 13310, timing group from output port.
[WARNING STA-0164] /OpenROAD-flow-scripts/flow/platforms/asap7/lib/asap7sc7p5t_SIMPLE_RVT_TT_nldm_211120.lib.gz line 13343, timing group from output port.
[WARNING STA-0164] /OpenROAD-flow-scripts/flow/platforms/asap7/lib/asap7sc7p5t_SIMPLE_RVT_TT_nldm_211120.lib.gz line 13376, timing group from output port.
[WARNING STA-0164] /OpenROAD-flow-scripts/flow/platforms/asap7/lib/asap7sc7p5t_SIMPLE_RVT_TT_nldm_211120.lib.gz line 14772, timing group from output port.
[WARNING STA-0164] /OpenROAD-flow-scripts/flow/platforms/asap7/lib/asap7sc7p5t_SIMPLE_RVT_TT_nldm_211120.lib.gz line 14805, timing group from output port.
[WARNING STA-0164] /OpenROAD-flow-scripts/flow/platforms/asap7/lib/asap7sc7p5t_SIMPLE_RVT_TT_nldm_211120.lib.gz line 14838, timing group from output port.
```

**Comment by rovinski** [comment]
They cannot, the warnings are due to the structure of the cells in the PDK and the way they were characterized into a timing model. You can, however, stop the warnings from showing up by using `suppress_message STA 164`.

**Comment by vijayank88** [comment]
I have discussed about this `WARNING STA-016` month back with Tom. Replied:
```
The output to output timing edges in the liberty cause this warning.
If it's just this one cell, you could add it to the don't use list. 
I think this warning means the timing will be less accurate.
If you are doing experiments it may also be safe to ignore.
```

**Reply by oharboe** [reply]
@tspyrou Given that ASAP7 never goes into production, can a default choice be made in ORFS that removes this warning?

I dont know how to translate the above advice into action...

**Reply by vijayank88** [reply]
@oharboe Add following lines in the [read_liberty.tcl](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/master/flow/scripts/read_liberty.tcl) beginning of the file. It will remove that warnings from the log.
```
if {[info exists ::env(PLATFORM)] && $::env(PLATFORM) == "asap7"} {
   suppress_message STA 164
}
```

**Reply by oharboe** [reply]
Thanks! Though what I'm wondering is if there is a best choice that be made on behalf of everyone using ORFS, so that this warning can be silenced in master.

I'm also concerned that the above suppresses other messages that I should investigate.

**Reply by vijayank88** [reply]
Ok. I will make a PR.

**Reply by vijayank88** [reply]
> Thanks! Though what I'm wondering is if there is a best choice that be made on behalf of everyone using ORFS, so that this warning can be silenced in master.
> 
> I'm also concerned that the above suppresses other messages that I should investigate.

That is why I restricted only to `ASAP7` platform.

**Reply by vijayank88** [reply]
PR created: https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/1056
